docs: update range query docs (#850)
GreptimeTeam/docs: docs/v0.7/en/reference/sql/range.md

   - Strings based on the `PromQL Time Durations` format (eg: `3h`, `1h30m`). Visit the [Prometheus documentation](https://prometheus.io/docs/prometheus/latest/querying/basics/#time-durations) for a more detailed description of this format.
   - `Interval` type. To use the `Interval` type, you need to carry parentheses, (for example: `(INTERVAL '1 year 3 hours 20 minutes')`). Visit [Interval](./functions.md#interval) for a more detailed description of this format.
- `AGGR_FUNCTION(column1, column2,..) RANGE INTERVAL [FILL FILL_OPTION]` is called a Range expression.

>    - `AGGR_FUNCTION(column1, column2,..)` is an aggregate function that represents the expression that needs to be aggregated. Aggregate functions support the use of `order by` expressions. (Note: `first_value` and `last_value` functions use timestamp as `order by` expression by default)

Reviewer: I don't understand what NOTE means. Does that mean the type of the column specified by `order by` must be a timestamp? Does that mean that other aggregate functions can use any type of column as an order by parameter?

Author: The `first_value` aggregate function is used to find the first value in an aggregation group. It generally needs to be used with `order by` to specify what the first value is. In range queries, timestamps are widely used as the sorting basis. Therefore, in order to simplify the writing of SQL, the default `order by` parameter of `first_value` is timestamp, and the same for `last_value`.

| `LINEAR` | Use [linear interpolation](https://en.wikipedia.org/wiki/Linear_interpolation) to fill the data. If an integer type is filled with `LINEAR`, the variable type of the column will be implicitly converted to a floating point type during calculation |
|   `X`    |                                           Fill in a constant, the data type of the constant must be consistent with the variable type of the Range expression                                            |

> :::tip NOTE

Reviewer: A note shouldn't be too long, and please avoid using too many notes.
Here seems to be a special case of the `FILL` option, so please remove the `:::tip NOTE` and put it at the bottom of the introduction of the `FILL` option.

> Note that if there are multiple Range expressions and the FILL method is specified for only one of them, in order to keep the number of SQL output rows uniform, the other Range expressions will use the FILL NULL method to fill in the missing time slots. So the following two SQL statements are equivalent in output:

Reviewer: ```suggestion
If there are multiple Range expressions but only one of them specifies the `Fill` option, the other Range expressions will use the `FILL NULL` method to fill in the missing time slots.
The following two SQL statements are equivalent:
```

```sql
SELECT 
    ts, 
    host, 
    min(cpu) RANGE '10s',
    max(cpu) RANGE '10s' FILL LINEAR 
FROM host_cpu 
ALIGN '5s';
```

equal to

```sql
SELECT 
    ts, 
    host, 
    min(cpu) RANGE '10s' FILL NULL,
    max(cpu) RANGE '10s' FILL LINEAR 
FROM host_cpu 
ALIGN '5s';
```
:::

Take the following table as an example:

```sql
+---------------------+-------+------+

> | ts                  | host  | val  |

Reviewer: If you are using `val` as the column name, please rename all `cpu` column to `val` column in the section of the `FILL` option, including SQLs and results.

+---------------------+-------+------+
| 1970-01-01 00:00:00 | host1 |    0 |
| 1970-01-01 00:00:15 | host1 |    6 |
| 1970-01-01 00:00:00 | host2 |    6 |

> | 1970-01-01 00:00:15 | host2 |   12 |

Reviewer: The example should have a row with value `NULL` to show the behavior of the different `FILL` options.

Author: It's the new behavior of FILL option, see pr https://github.com/GreptimeTeam/greptimedb/pull/3489

+---------------------+-----------------------------------+
```

## `ORDER BY` option

> Range queries support the use of `order by` expressions within specific aggregate functions to sort data on the same time slot. Taking the `first_value` and `last_value` aggregate functions as an example, users can use `order by` to sort the aggregated data to correctly select the first value and the last value. By default, the `first_value` and `last_value` functions sort data in ascending order using timestamps.

Reviewer: ```suggestion
Range queries support the use of `order by` expressions in the parameters of the `first_value` and `last_value` aggregate functions. By default, the data is sorted in ascending order based on the time index column.
```

Take this table as an example:

```sql
+---------------------+-------+------+-------+
| ts                  | host  | val  | addon |
+---------------------+-------+------+-------+
| 1970-01-01 00:00:00 | host1 |    0 |     3 |
| 1970-01-01 00:00:01 | host1 |    1 |     2 |
| 1970-01-01 00:00:02 | host1 |    2 |     1 |
+---------------------+-------+------+-------+
```

> Users can not specify the `order by` expression, and the effect is equivalent to using the `ts` column to sort in ascending order.

Reviewer: ```suggestion
When the `order by` expression is not specified in the function parameter, the default behavior is to sort the data in ascending order based on the time index column.
```

```sql
SELECT ts, first_value(val) RANGE '5s', last_value(val) RANGE '5s' FROM host ALIGN '5s';
-- Equivalent to
SELECT ts, first_value(val order by ts ASC) RANGE '5s', last_value(val order by ts ASC) RANGE '5s' FROM host ALIGN '5s';
```

Get after query

```sql
+---------------------+--------------------------------+-------------------------------+
| ts                  | FIRST_VALUE(host.val) RANGE 5s | LAST_VALUE(host.val) RANGE 5s |
+---------------------+--------------------------------+-------------------------------+
| 1970-01-01 00:00:00 |                              0 |                             2 |
+---------------------+--------------------------------+-------------------------------+
```

> Users can also specify their own sorting rules without using the default sorting options, such as using `addon` sorting:

Reviewer: ```suggestion
You can specify your own sorting rules. For example, the following SQL sorts the data by the `addon` column in ascending order:
```

```sql
SELECT ts, first_value(val ORDER BY addon ASC) RANGE '5s', last_value(val ORDER BY addon ASC) RANGE '5s' FROM host ALIGN '5s';
```

Get after query

```sql
+---------------------+---------------------------------------------------------------------+--------------------------------------------------------------------+
| ts                  | FIRST_VALUE(host.val) ORDER BY [host.addon ASC NULLS LAST] RANGE 5s | LAST_VALUE(host.val) ORDER BY [host.addon ASC NULLS LAST] RANGE 5s |
+---------------------+---------------------------------------------------------------------+--------------------------------------------------------------------+
| 1970-01-01 00:00:00 |                                                                   2 |                                                                  0 |
+---------------------+---------------------------------------------------------------------+--------------------------------------------------------------------+
```

> 

Reviewer: What is the behavior of the following SQL:
```
SELECT ts, first_value(val ORDER BY addon ASC) RANGE '5s', last_value(val ORDER BY addon ASC) RANGE '5s' FROM host ALIGN '5s' ORDER BY addon DESC;
```
ORDER BY also added at the end of the SQL and the specified behavior is different from the former.

Reviewer: What is the behavior of the SQL that only specified `order by` at the end of the SQL
```
SELECT ts, first_value(val) RANGE '5s', last_value(val) RANGE '5s' FROM host ALIGN '5s' ORDER BY addon DESC;
```

Author: - `first_value(val ORDER BY addon ASC) RANGE '5s'` means sort data in aggragate group, so we can choose our desire value in `first_value`.
- order by in SQL level meas sort the output of entire output data.
